res/controllers/common-controller-scripts.js

```diff
@@ -121,616 +102,513 @@ var msecondstominutes = function(msecs) {
+     *
+     * Returns true when was an object was created using the `{}` synthax.
+     * False if the object is an instance of a class like Date or Proxy or an Array.
+     *
+     * isSimpleObject({}) // true
+     * isSimpleObject(null) // false
+     * isSimpleObject(undefined) // false
+     * isSimpleObject(new Date) // false
+     * isSimpleObject(new (class {})()) // false
+     * @param {any} obj Object to test
+     * @returns {boolean} true if obj was created using the `{}` or `new Object()` synthax, false otherwise
+     */
+    isSimpleObject: function(obj) {
+        return obj !== null && typeof obj === "object" && obj.constructor.name === "Object";
+    },
+
+
+    /**
+     * Deeply merges 2 objects (Arrays and Objects only, not Map for instance).
+     * @param target {object | Array} Object to merge source into
+     * @param source {object | Array} Object to merge into source
+     */
+    deepMerge: function(target, source) {
+        if (target === source || target === undefined || target === null || source === undefined || source === null) {
+            return;
+        }
 
-    if (Array.isArray(target) && Array.isArray(source)) {
-        const objTarget = target.reduce((acc, val, idx) => Object.assign(acc, {[idx]: val}), {});
-        const objSource = source.reduce((acc, val, idx) => Object.assign(acc, {[idx]: val}), {});
-        deepMerge(objTarget, objSource);
-        target.length = 0;
-        target.push(...Object.values(objTarget));
-    } else if (isSimpleObject(target) && isSimpleObject(source)) {
-        Object.keys(source).forEach(key => {
-            if (
-                Array.isArray(target[key]) && Array.isArray(source[key]) ||
-              isSimpleObject(target[key]) && isSimpleObject(source[key])
-            ) {
-                deepMerge(target[key], source[key]);
-            } else if (source[key] !== undefined && source[key] !== null) {
-                Object.assign(target, {[key]: source[key]});
+        if (Array.isArray(target) && Array.isArray(source)) {
+            const objTarget = target.reduce((acc, val, idx) => Object.assign(acc, {[idx]: val}), {});
+            const objSource = source.reduce((acc, val, idx) => Object.assign(acc, {[idx]: val}), {});
+            this.deepMerge(objTarget, objSource);
+            target.length = 0;
+            target.push(...Object.values(objTarget));
+        } else if (this.isSimpleObject(target) && this.isSimpleObject(source)) {
+            Object.keys(source).forEach(key => {
+                if (
+                    Array.isArray(target[key]) && Array.isArray(source[key]) ||
+                    this.isSimpleObject(target[key]) && this.isSimpleObject(source[key])
+                ) {
+                    this.deepMerge(target[key], source[key]);
+                } else if (source[key] !== undefined && source[key] !== null) {
+                    Object.assign(target, {[key]: source[key]});
+                }
+            });
+        }
```

Comment:
as this is not a class, `this` is not `script` but instead the function object. You'll have to call `script.deepMerge/isSimpleObject` explicitly.

Reply:
I tried both, no difference in the error message.

Reply:
interesting. Well either way, `this` is wrong.